web/src/compose_paste.ts

```diff
@@ -434,7 +434,7 @@ export function paste_handler_converter(
                 (href = node.firstElementChild.getAttribute("href")) === null ||
                 !copied_html
                     .querySelector("a[href='" + CSS.escape(href) + "']")
-                    ?.parentElement?.classList.contains("message_inline_image")
+                    ?.parentElement?.classList.contains("message-linked-media")
```

Comment:
In this file, it might be nice to support `message_inline_image` too, since we may be processing content pasted from an older Zulip version.

Reply:
Yes. I've pulled the changes to the paste handler out into their own commit, as I have made some adjustments to the logic that will need careful review.